File: day1/cpp/part1.cpp
#include <fstream>
#include <iostream>

using namespace std;

int main() {
    ifstream f("input.txt");
    string s;
    int sum{};
    int mx{};
    while (getline(f, s)) {
        if (s != "") {
            sum += stoi(s);
            if (sum > mx)
                mx = sum;
        } else
            sum = 0;
    }
    cout << mx << endl;
    f.close();
}

File: 2022/day1/elixir/part1.exs
{:ok, file} = File.open("../input.txt", [:read])

contents = IO.read(file, :all) |> String.split("\n\n")

lists =
  Enum.map(contents, fn section ->
    section
    |> String.split("\n")
    |> Enum.map(&String.to_integer/1)
    |> Enum.sum()
  end)

Enum.max(lists) |> IO.puts()

File.close(file)
